src/session/session_api.c

```diff
@@ -149,6 +149,10 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
 
 	SESSION_API_CALL(session, alter, config, cfg);
 
+	/* In-memory ignores alter operations. */
+	if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
```

Comment:
I'm OK with this, since it applies to all the configuration options currently exposed via alter - I expect it to change in the future when we expose an alter option that makes sense for in-memory configuration. We can resolve that when it arises.